File: JobSearch-SPA/src/app/_models/Application.ts
import { Advertisment } from './Advertisment';
import { User } from './User';

export class Application {

    applicationId: number;
    applicationDate : Date;
    description : string;
    cv: string;
    motivation: string;

    user: User;
    advertisment: Advertisment;

    advertismentId?: number;
    userId?: number;
}

File: JobSearch.WebAPI/Database/Application.cs
using System;
using System.Collections.Generic;

namespace JobSearch.WebAPI.Database
{
    public partial class Application
    {
        public int ApplicationId { get; set; }
        public DateTime ApplicationDate { get; set; }
        public string Description { get; set; }
        public string Cv { get; set; }
        public string Motivation { get; set; }
        public int? AdvertismentId { get; set; }
        public int? UserId { get; set; }

        public virtual Advertisment Advertisment { get; set; }
        public virtual Users User { get; set; }
    }
}
